aait/backend/group-2B/Application/Features/Posts/Handlers/Commands/CreatePostCommandHandler.cs

```diff
@@ -0,0 +1,47 @@
+using AutoMapper;
+using MediatR;
+using SocialSync.Application.Contracts.Persistence;
+using SocialSync.Application.DTOs.PostDtos;
+using SocialSync.Application.DTOs.PostDtos.Validators;
+using SocialSync.Application.Features.Posts.Requests.Commands;
+using SocialSync.Application.Common.Responses;
+using SocialSync.Domain.Entities;
+
+namespace SocialSync.Application.Features.Posts.Handlers.Commands;
+
+public class CreatePostCommandHandler : PostsRequestHandler, IRequestHandler<CreatePostCommand, BaseCommandResponse>
+{
+    public CreatePostCommandHandler(IUnitOfWork unitOfWork, IMapper mapper) : base(unitOfWork, mapper)
+    {
+    }
+
+    public async Task<BaseCommandResponse> Handle(CreatePostCommand request, CancellationToken cancellationToken)
+    {
+
+        var CreateValidator = new CreatePostDtoValidator(_userRepository);
+        var ValidationResult = await CreateValidator.ValidateAsync(request.CreatePostDto);
+        var response = new BaseCommandResponse();
+
+        if (!ValidationResult.IsValid)
+        {
+
+            response.Success = false;
+            response.Message = $"Post Creation Failed";
+            response.Errors = ValidationResult.Errors.Select(q => q.ErrorMessage).ToList();
+
+        }
+        else
+        {
+
+            var post = _mapper.Map<Post>(request.CreatePostDto);
+            var newPost = await _postRepository.AddAsync(post);
+
```

Comment:
You have to save the changes to the database and check the result if it is successfull